app/models/grda_warehouse/hud/client.rb

```diff
@@ -406,6 +406,31 @@ class << self
       where(id: GrdaWarehouse::ClientFile.consent_forms.confirmed.pluck(:client_id))
     end
 
+    scope :viewable_by, -> (user) do
+      if user.can_edit_anything_super_user?
+        current_scope
+      elsif user.coc_codes.none?
+        none
+      else
+        at = arel_table
+        e_t = GrdaWarehouse::Hud::Enrollment.arel_table
+        e_coc_t = GrdaWarehouse::Hud::EnrollmentCoc.arel_table
+        where(
+          e_t.project(Arel.star).
```

Comment:
This is already a has, many, is there anyway to use the existing join logic?

Reply:
Yes, using ActiveRecord joins simpler and prevents your forgetting to handle soft deletion, as I did here. It's just that because it's a has-many relation, you need to use `distinct` or something and I've had trouble with the composability of those in the past. Anyway, I'll do that and figure if we have composition problems in the future we'll deal with them then.

Reply:
Incidentally, the exists subquery is about 33% faster. Maybe leaving out the soft deletion logic has something to do with this.